src/include/duckdb.h

```diff
@@ -162,6 +171,7 @@ typedef struct {
 	char *__deprecated_name;
 #endif
 	void *internal_data;
+	void *duckdb_logical_type;
```

Comment:
The current idea with the deprecated flags is that we migrate to storing all of the data for the `duckdb_column` class inside the `internal_data` object. This is cleaner since it means we don't need to alter a struct in the header to add new fields. Altering a struct in the header is problematic since it breaks backwards compatibility during linking, as the sizes of objects on e.g. the stack changes which can cause nasty crashes. 

The `internal_data` object can be a C++ class that holds C++ objects, and should be defined in `src/include/duckdb/main/capi_internal.hpp`. For example, similarly to the `DatabaseData` and `AppenderWrapper` objects, there should be a `DuckDBColumnData` object.

This object can then hold the LogicalType as-is, and adding new fields to this will be easy as well.

Could you perhaps have a look at creating the `internal_data` object like this and using it to store the LogicalType?

Reply:
OK

Reply:
I had move LogicalType into internal_data.

What's our plan to the __deprecated* fields? Should I also define them in the internal_data?

Reply:
For now the deprecated fields can remain in the header. I am planning on doing some restructuring of the C API in the future which includes removing the deprecated fields and moving them into the internal_data amongst others, which is why they are deprecated.